app/org/catalyst/slackservice/controllers/AuthController.java

```diff
@@ -43,12 +44,21 @@ public AuthController(AppService service, AppConfig config, MessagesApi messages
      */
     public CompletionStage<Result> handle(Http.Request httpRequest) {
         var requestCode = httpRequest.queryString("code");
+        var error = httpRequest.queryString("error");
 
-        if (requestCode.isEmpty()) {
+        if (requestCode.isEmpty() && error.isEmpty()) {
             var messages = new MessageHandler(_messagesApi.preferred(httpRequest));
             return ResultHelper.badRequest(messages, MessageHandler.MISSING_CODE);
         }
 
+        if (error.isPresent()) {
```

Comment:
What is the reasoning behind using `isEmpty` and `isPresent` instead of `isEmpty` and `!isEmpty`?  Are `isEmpty` and `isPresent` inverses of each other, could any case be left unhandled?

Reply:
Looking through the documentation of Optional, they're returning the opposite of each other. isPresent looks more readable, imo. Now that I think about it, I'll probably just move up the error.isPresent to avoid the double condition check.